src/Avalonia.Base/Rendering/SceneGraph/DrawOperation.cs

```diff
@@ -11,7 +11,10 @@ internal abstract class DrawOperation : IDrawOperation
     {
         public DrawOperation(Rect bounds, Matrix transform)
         {
-            bounds = bounds.Normalize().TransformToAABB(transform);
+            bounds = bounds
```

Comment:
I'm not quite sure it's a good idea to inflate every single draw operation, especially plain rects that are used as backgrounds. That might cause **way more** elements to be drawn in a particular frame.

Reply:
I agree, this whole PR is unfortunately a hack and probably isn't a good idea. It's better to find and fix the rounding errors.

Reply:
I somewhat agree with both of you. But I want to know if that fix would at least solve the issue. Maybe only some geometries needs this fix. After the PR is identified as a valid solution, the cases where this is needed can be refined.